pkg/discovery/network/generic.go

```diff
@@ -153,50 +162,23 @@ func parseServiceCIDRFrom(msg string) (string, error) {
 }
 
 func findPodIPRange(ctx context.Context, client controllerClient.Client) (string, error) {
-	podIPRange, err := findPodIPRangeKubeController(ctx, client)
+	podIPRange, err := findPodIPRangeFromKubeController(ctx, client)
 	if err != nil || podIPRange != "" {
 		return podIPRange, err
 	}
 
-	podIPRange, err = findPodIPRangeKubeProxy(ctx, client)
-	if err != nil || podIPRange != "" {
-		return podIPRange, err
-	}
-
-	podIPRange, err = findPodIPRangeFromNodeSpec(ctx, client)
+	podIPRange, err = findPodIPRangeFromKubeProxy(ctx, client)
 	if err != nil || podIPRange != "" {
 		return podIPRange, err
 	}
 
 	return "", nil
 }
 
-func findPodIPRangeKubeController(ctx context.Context, client controllerClient.Client) (string, error) {
+func findPodIPRangeFromKubeController(ctx context.Context, client controllerClient.Client) (string, error) {
 	return FindPodCommandParameter(ctx, client, "component=kube-controller-manager", "--cluster-cidr")
 }
 
-func findPodIPRangeKubeProxy(ctx context.Context, client controllerClient.Client) (string, error) {
-	return FindPodCommandParameter(ctx, client, "component=kube-proxy", "--cluster-cidr")
-}
-
-func findPodIPRangeFromNodeSpec(ctx context.Context, client controllerClient.Client) (string, error) {
-	nodes := &corev1.NodeList{}
-
-	err := client.List(ctx, nodes)
-	if err != nil {
-		return "", errors.WithMessagef(err, "error listing nodes")
-	}
-
-	return parseToPodCidr(nodes.Items)
-}
-
-func parseToPodCidr(nodes []corev1.Node) (string, error) {
-	// In K8s, each node is typically assigned a unique PodCIDR range for the pods that run on that node.
-	// Each node's PodCIDR is used to allocate IP addresses to the pods scheduled on that node. Only if
-	// the cluster is a single node deployment, we should rely on the node.Spec.PodCIDR as podCIDR of the cluster.
-	if len(nodes) == 1 {
-		return nodes[0].Spec.PodCIDR, nil
-	}
```

Comment:
From the PR summary:

> The node.spec.podCIDR field is not the cluster pod ip range, but the node-assigned.

But this is only considered if there's only one node as explained in the comments above and if the pod CIDR wasn't found from any prior methods. It seems valid to keep this.

Reply:
You are right. I have fixed it.